geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/distributed/LuceneQueryFunction.java

```diff
@@ -122,6 +128,98 @@ public void execute(FunctionContext<LuceneFunctionContext> context) {
         | PrimaryBucketException e) {
       logger.debug("Exception during lucene query function", e);
       throw new InternalFunctionInvocationTargetException(e);
+    } catch (LuceneIndexCreationInProgressException ex) {
+      handleException(ex, ctx);
+    }
+  }
+
+  private void handleException(LuceneIndexCreationInProgressException ex,
+      RegionFunctionContext ctx) {
+    PartitionedRegion userDataRegion = (PartitionedRegion) ctx.getDataSet();
+    // get local version
+    Version localVersion = ((InternalCache) ctx.getCache()).getDistributionManager()
+        .getDistributionManagerId().getVersionObject();
+    // get the remote members
+    Set<InternalDistributedMember> remoteMembers =
+        userDataRegion.getRegionAdvisor().adviseAllPRNodes();
+
+    if (!remoteMembers.isEmpty()) {
+      for (InternalDistributedMember remoteMember : remoteMembers) {
+        if (remoteMember.getVersionObject().ordinal() < localVersion.ordinal()) {
```

Comment:
I think you will want to look for an explicit version here.  In the future, we would only want to "wait" if the version was pre 1.6.  For example  1.8 to 1.7 should not wait.

Reply:
pre 1.6 or including 1.6?

Reply:
Agree with Jason.